core/src/main/scala/scalafix/syntax/package.scala

```diff
@@ -25,6 +37,51 @@ package object syntax {
       if (config.hasPath(key)) config.getBoolean(key)
       else els
   }
+  implicit class XtensionCompleted[T](completed: Completed[T]) {
+    // TODO(olafur) contribute these upstream
+    def toOption: Option[T] = completed match {
+      case Completed.Success(e) => Some(e)
+      case _ => None
+    }
+    def toEither: Either[Exception, T] = completed match {
+      case Completed.Success(e) => Right(e)
+      case Completed.Error(e) => Left(e)
+    }
+    def toTry: Try[T] = Try(completed.get)
+  }
+  implicit class XtensionSymbol(symbol: Symbol) {
+    /** Returns simplified version of this Symbol.
+      *
+      * - No Symbol.Multi
+      * - No Signature.{Type,Method}
+      */
+    def normalized: Symbol = symbol match {
+      case Symbol.Multi(sym +: _) => sym.normalized
+      case Symbol.Global(sym, Signature.Type(name)) =>
+        Symbol.Global(sym, Signature.Term(name))
+      case Symbol.Global(Symbol.Global(sym, Signature.Term(name)),
+                         Signature.Method("apply", _)) =>
+        Symbol.Global(sym, Signature.Term(name))
+      case Symbol.Global(sym, Signature.Method(name, _)) =>
+        Symbol.Global(sym.normalized, Signature.Term(name))
+      case x => x
+    }
+
+    /** Returns corresponding scala.meta.Tree for symbol
+      * Caveat: not thoroughly tested, may crash
+      * */
+    def to[T <: Tree]: Try[T] = Try {
+      val tree = symbol match {
+        case Symbol.Global(Symbol.None, Signature.Term(name)) =>
+          Term.Name(name)
+        case Symbol.Global(sym, Signature.Type(name)) =>
+          Type.Select(sym.to[Term.Ref].get, Type.Name(name))
+        case Symbol.Global(sym, Signature.Term(name)) =>
+          Term.Select(sym.to[Term].get, Term.Name(name))
+      }
+      tree.asInstanceOf[T] // should crash
+    }
+  }
```

Comment:
That would be perfect 👍 I opened a ticket in scala.meta https://github.com/scalameta/scalameta/issues/675